server/src/main/java/org/apache/iotdb/db/newsync/pipedata/PipeData.java

```diff
@@ -53,39 +58,30 @@ public long getSerialNumber() {
 
   public long serialize(DataOutputStream stream) throws IOException {
     long serializeSize = 0;
-    stream.writeLong(serialNumber);
-    serializeSize += Long.BYTES;
     stream.writeByte((byte) getType().ordinal());
     serializeSize += Byte.BYTES;
-    serializeSize += serializeImpl(stream);
+    stream.writeLong(serialNumber);
+    serializeSize += Long.BYTES;
     return serializeSize;
   }
 
-  public abstract long serializeImpl(DataOutputStream stream) throws IOException;
-
   public static PipeData deserialize(DataInputStream stream)
       throws IOException, IllegalPathException {
-    long serialNumber = stream.readLong();
     Type type = Type.values()[stream.readByte()];
-
-    PipeData pipeData = null;
     switch (type) {
       case TSFILE:
-        pipeData = new TsFilePipeData(serialNumber);
-        break;
+        return TsFilePipeData.deserialize(stream);
       case DELETION:
-        pipeData = new DeletionPipeData(serialNumber);
-        break;
+        return DeletionPipeData.deserialize(stream);
       case PHYSICALPLAN:
-        pipeData = new SchemaPipeData(serialNumber);
-        break;
+        return SchemaPipeData.deserialize(stream);
+      default:
+        logger.error("Deserialize PipeData error because Unknown type {}.", type);
```

Comment:
throw a UnsupportException may be better

Reply:
fixed